### PR TITLE
Singularities

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp [hunk 1]
```diff
@@ -115,7 +120,10 @@ TEST_F(EmbeddedExplicitRungeKuttaNyströmIntegratorTest,
       std::bind(HarmonicOscillatorToleranceRatio,
                 _1, _2, length_tolerance, speed_tolerance, step_size_callback);
 
-  integrator.Solve(problem, adaptive_step_size);
+  {
```
Comment (pleroy) [inline integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp]: I don't like these blocks.

File: integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp [hunk 2]
```diff
@@ -223,5 +238,67 @@ TEST_F(EmbeddedExplicitRungeKuttaNyströmIntegratorTest,
   }
 }
 
+TEST_F(EmbeddedExplicitRungeKuttaNyströmIntegratorTest,
+       Singularity) {
```
Comment (pleroy) [inline integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp]: Fits on the previous line?

File: integrators/ordinary_differential_equations.hpp [hunk 3]
```diff
@@ -132,6 +132,12 @@ class FixedStepSizeIntegrator : public Integrator<DifferentialEquation> {
   serialization::FixedStepSizeIntegrator::Kind const kind_;
 };
 
+enum class TerminationCondition {
+  Done,
+  MaxSteps,
```
Comment (pleroy) [inline integrators/ordinary_differential_equations.hpp]: Maybe more explicit: MaximumStepsReached?

File: integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp (hunk 2 above)
Comment (eggrobin) [inline integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp]: Yes.

File: integrators/ordinary_differential_equations.hpp (hunk 3 above)
Comment (eggrobin) [inline integrators/ordinary_differential_equations.hpp]: `ReachedMaximalStepCount`

File: integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp (hunk 1 above)
Comment (eggrobin) [inline integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp]: :'(

**Pull request merged**